### PR TITLE
Jamarw/link ui changes

### PR DESCRIPTION
**Pull request by jamarw**
1. Create new dropdown option for internalLink command  (link to file in repo).
2. Create new dropdown option for insertBookmark command (link to heading).
3. Update selectLinkType function to only list heading (bookmark) commands.
4. Rename output stream.
5. Bump vscode engine version.
6. Adjust active editor checks.
7. Add new function to let user know that there is no active editor/tabs open to modify.

### CODE REVIEW

File: docs-markdown/src/controllers/preview-controller.ts [hunk 1]
```diff
@@ -17,25 +17,29 @@ export function previewTopic() {
...
+        const osPlatform = common.getOSPlatform();
+        const extensionName = "docsmsft.docs-preview";
+        const { msTimeValue } = common.generateTimestamp();
+        const friendlyName = "docsmsft.docs-preview".split(".").reverse()[0];
+        const inactiveMessage = `[${msTimeValue}] - The ${friendlyName} extension is not installed.`;
+        if (common.checkExtension(extensionName, inactiveMessage)) {
+            if (osPlatform === "win32") {
+                vscode.commands.executeCommand("docs.showPreviewToSide");
+            } else {
+                vscode.commands.executeCommand("markdown.showPreviewToSide");
```
Comment (qinezh) [inline docs-markdown/src/controllers/preview-controller.ts]: docs-preview supports cross-platform, do we still need to use vscode default markdown preview in Linux/macOS?

File: docs-markdown/src/controllers/table-controller.ts [hunk 2]
```diff
@@ -17,36 +17,45 @@ export function insertTableCommand() {
...
-            log.debug("Table insert failed.");
+        if (!common.isMarkdownFileCheck(editor, false)) {
+            return;
         }
-    });
+
+        const tableInput = vscode.window.showInputBox({ prompt: "Input the number of columns and rows as C:R" });
+
+        // gets the users input on number of columns and rows
+        tableInput.then((val) => {
```
[the remark this answers is not in the document]
Comment (jamarw) [reply docs-markdown/src/controllers/table-controller.ts]: Good call, just here for linting.

File: docs-markdown/src/helper/common.ts [hunk 3]
```diff
@@ -277,12 +281,13 @@ export function generateTimestamp() {
...
-    try {
-        return vscode.extensions.getExtension(extensionName).isActive;
-    } catch (error) {
+    const extensionValue = vscode.extensions.getExtension(extensionName);
+    if (!extensionValue) {
         if (notInstalledMessage) {
             output.appendLine(notInstalledMessage);
         }
         return false;
+    } else {
```
[the remark this answers is not in the document]
Comment (jamarw) [reply docs-markdown/src/helper/common.ts]: Fixed

File: docs-markdown/src/controllers/snippet-controller.ts [hunk 4]
```diff
@@ -20,58 +22,67 @@ export function insertSnippetCommand() {
...
         scopeOptions.push({ label: "Scoped Search", description: "Look in specific directories for snippet" });
 
         vscode.window.showQuickPick(scopeOptions).then(function searchType(selection) {
-            const searchSelection = selection.label;
-
-            if (searchSelection === "Full Search") {
-                utilityHelper.search(editor, selected, searchTerm, folderPath);
+            if (!selection) {
+                return;
             } else {
```
[the remark this answers is not in the document]
Comment (jamarw) [reply docs-markdown/src/controllers/snippet-controller.ts]: Fixed

File: docs-markdown/src/controllers/preview-controller.ts (hunk 1 above)
Comment (qinezh) [reply docs-markdown/src/controllers/preview-controller.ts]: @wibjorn I mean `docs-preview` has already supported cross-platform, you could give it a try :)

File: docs-markdown/src/controllers/snippet-controller.ts [hunk 5]
```diff
@@ -20,58 +22,67 @@ export function insertSnippetCommand() {
...
-    // We need a file to search for, calls the searchRepo function.
     vscode.window.showInputBox({ prompt: "Enter snippet search terms." }).then(searchRepo);
-
-    // finds the directories to search, passes this and the search term to the search function.
-    function searchRepo(searchTerm: any) {
+}
+// finds the directories to search, passes this and the search term to the search function.
+export function searchRepo(searchTerm: any) {
+    const editor = vscode.window.activeTextEditor;
+    if (!editor) {
```
[the remark this answers is not in the document]
Comment (jamarw) [reply docs-markdown/src/controllers/snippet-controller.ts]: Fixed

**Comment by wibjorn** [comment]
Looking good, let's just remember to try debugging the "Link to Heading" > "Heading in this file" quick pick spacing after this merges.

**Pull request merged**